File: algorithms/algorithms-cplusplus/leetcode/Question_0128_Longest_Consecutive_Sequence.cc
#include <vector>
#include <unordered_set>

using namespace std;

class Solution {
 public:
  int longestConsecutive(vector<int>& nums) {
    if (nums.size() <= 1) {
      return nums.size();
    }
    unordered_set<long> s(nums.begin(), nums.end());
    int res = 0, cnt = 0;
    for (long n : nums) {
      if (!s.count(n - 1)) {
        cnt = 0;
        while (s.count(n++)) {
          cnt++;
        }
        res = max(res, cnt);
      }
    }
    return res;
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0128_Longest_Consecutive_Sequence.java
package com.lanceolata.coding.algorithms.java.leetcode;

import java.util.HashSet;
import java.util.Set;

public class Question_0128_Longest_Consecutive_Sequence {
    public int longestConsecutive(int[] nums) {
        if (nums == null) {
            return 0;
        }
        Set<Integer> set = new HashSet<>();
        for (int n : nums) {
            set.add(n);
        }
        int res = 0;
        for (int n : nums) {
            if (!set.contains(n - 1)) {
                int cnt = 0;
                while (set.contains(n)) {
                    cnt++;
                    n++;
                }
                res = Math.max(res, cnt);
            }
        }
        return res;
    }
}
